### DISCUSSION TITLE
delete images from collection

### DISCUSSION THREAD

**Discussion by yas1n3** [Richard-Weiss/Bing-Creator-Image-Downloader#48] [Ideas]
Right now, the script can delete a collection after download, but is it possible to delete individual images from collection once the download is successful in a future commit

**Comment by Richard-Weiss** [comment]
@yas1n3 If you have some specific requirements, sure.

**Reply by yas1n3** [reply]
No not really, I wanted to add that feature but I have no idea which endpoint is responsible for deleting the images from the collection, I might give it a try but no success is guaranteed.  I wish MS could open up their api for people like us.

**Reply by Richard-Weiss** [reply]
@yas1n3 
Oh, I see. You can look at the [`delete_collection` function in the `CollectionUtility` class](https://github.com/Richard-Weiss/Bing-Creator-Image-Downloader/blob/main/utilities/collection_utility.py).  
I try to make my code easy to understand, but feel free to ask me any questions regarding it.  
That endpoint will delete the whole collection though. 
The endpoint for deleting single images is `https://www.bing.com/mysaves/collections/items/delete?sid=0` with the body looking like this:
```
{
    "ItemReferences": [],
    "Items": [
        {
            "Id": "/prism/imagecreator/v1/image:$sicId",
            "SourceCollectionId": "$collectionId"
        }
    ]
}
```
You can get the sicId through the details API for example. It's different than the imageId.

**Reply by yas1n3** [reply]
Thanks a lot, will look into it!!!
